src/main/java/com/iota/iri/service/API.java

```diff
@@ -531,8 +531,16 @@ public static void incEllapsedTime_getTxToApprove(long ellapsedTime) {
         Hash referenceHash = null;
         if(reference != null) {
             referenceHash = new Hash(reference);
-            if(!TransactionViewModel.exists(instance.tangle, referenceHash)) {
+            if (!TransactionViewModel.exists(instance.tangle, referenceHash)) {
+                //reference not found
                 referenceHash = null;
+            } else {
+                TransactionViewModel transactionViewModel = TransactionViewModel.fromHash(instance.tangle, referenceHash);
+                if (transactionViewModel.snapshotIndex() != 0
+                        && transactionViewModel.snapshotIndex() < instance.milestone.latestSolidSubtangleMilestoneIndex - depth) {
```

Comment:
Depth needs to be bounded else it can be abused.

Reply:
Depth is already bounded.